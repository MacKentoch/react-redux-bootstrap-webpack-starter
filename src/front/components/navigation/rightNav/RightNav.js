// @flow

// #region imports
import React from 'react';
import RightNavButton from './rightNavButton/RightNavButton';
// #endregion

// #region flow types
export type RightLink = {
  link: string,
  label: string,
  viewName: string,
};

export type OnRightNavButtonClick = (
  event: SyntheticEvent<>,
  viewName: string,
) => any;

type Props = {
  rightLinks: Array<RightLink>,
  onRightNavButtonClick: OnRightNavButtonClick,

  ...any,
};
// #endregion

const RightNav = ({ rightLinks, onRightNavButtonClick }: Props) => {
  return (
    <ul className="nav navbar-nav navbar-right">
      {rightLinks.map(({ link, label, viewName }, index) => {
        return (
          <RightNavButton
            key={index}
            link={link}
            label={label}
            viewName={viewName}
            onClick={onRightNavButtonClick}
          />
        );
      })}
    </ul>
  );
};

// #region statics
RightNav.defaultProps = { rightLinks: [] };
RightNav.displayName = 'RightNav';
// #endregion

export default RightNav;
