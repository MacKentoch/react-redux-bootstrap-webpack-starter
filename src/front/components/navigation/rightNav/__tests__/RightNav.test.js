// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import RightNav from '../RightNav';
import { type RightLink, type OnRightNavButtonClick } from '../RightNav';
// #endregion

// #region flow types
type Props = {
  rightLinks?: Array<RightLink>,
  onRightNavButtonClick?: OnRightNavButtonClick,
};
// #endregion

describe('RightNav component', () => {
  it('renders as expected', () => {
    const props: Props = {
      rightLinks: [
        { link: '1', label: 'view 1', viewName: 'view 1' },
        { link: '2', label: 'view 2', viewName: 'view 2' },
      ],
      onRightNavButtonClick: () => {},
    };

    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <RightNav {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
