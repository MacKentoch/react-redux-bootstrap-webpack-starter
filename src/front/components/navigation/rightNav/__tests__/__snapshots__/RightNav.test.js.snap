// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightNav component renders as expected 1`] = `
<div>
  <ul
    className="nav navbar-nav navbar-right"
  >
    <li>
      <a
        href="1"
        onClick={[Function]}
      >
        view 1
      </a>
    </li>
    <li>
      <a
        href="2"
        onClick={[Function]}
      >
        view 2
      </a>
    </li>
  </ul>
</div>
`;
